cmd/network/governance/proposal.go

```diff
@@ -0,0 +1,66 @@
+package governance
+
+import (
+	"os"
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/connection"
+
+	"github.com/oasisprotocol/cli/cmd/common"
+	cliConfig "github.com/oasisprotocol/cli/config"
+	governance "github.com/oasisprotocol/oasis-core/go/governance/api"
+)
+
+var propShowCmd = &cobra.Command{
+	Use:     "proposal-show <proposal-id>",
+	Short:   "Show governance proposal",
+	Args:    cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		cfg := cliConfig.Global()
+		npa := common.GetNPASelection(cfg)
+
+		// Determine the proposal ID to query
+		proposalID, err := strconv.ParseUint(args[0], 10, 64)
+		cobra.CheckErr(err)
+
+		// When not in offline mode, connect to the given network endpoint
+		ctx := context.Background()
+		conn, err := connection.Connect(ctx, npa.Network)
+		cobra.CheckErr(err)
+
+		// Get current height
+		height := common.GetHeight()
+		
+		// Get the proposal
+		proposalQuery := &governance.ProposalQuery {
+			Height:     height,
+			ProposalID: proposalID,
+		}
+		proposal, err := conn.Consensus().Governance().Proposal(ctx, proposalQuery)
+		cobra.CheckErr(err)
+		
+		// Write output
+		switch {
+			case proposal.Content.Upgrade != nil:
+				fmt.Println("Proposal type: Upgrade")
+				proposal.Content.Upgrade.PrettyPrint(ctx, "", os.Stdout)
+			case proposal.Content.CancelUpgrade != nil:
+				fmt.Println("Proposal type: Cancel Upgrade")
+				proposal.Content.CancelUpgrade.PrettyPrint(ctx, "", os.Stdout)
+			case proposal.Content.ChangeParameters != nil:
+				fmt.Println("Proposal type: Change Parameters")
+				proposal.Content.ChangeParameters.PrettyPrint(ctx, "", os.Stdout)
+			default:
+				fmt.Println("unknown")
+			}
```

Comment:
I don't get this, why don't you simply do:
```go
proposal.Content.PrettyPrint(ctx, "", os.Stdount)
```

Reply:
So that it can write proposal type in human friendly language. I can change it if you want.

Reply:
Doesn't the other one do that as well? It should write something like:
```
Upgrade:
  ...
```
And you can add something like `"Proposal content:"` before this and specify prefix as `"  "`.

Reply:
You can also update the `PrettyPrint` function in oasis-core if you think it would be nicer to also print out the "Proposal type: " part.